vendor/spawn/src/spawn_stubs.c

```diff
@@ -13,6 +13,39 @@ CAMLextern int caml_convert_signal_number(int);
 
 #if defined(__APPLE__)
 
+# if defined(__MAC_OS_X_VERSION_MAX_ALLOWED)
```

Comment:
@rgrinberg What this macro is supposed to do specifically?

Reply:
Yeah, this is wrong. The check should be for 10.5 because we need pthread_chdir

Reply:
I guess this is what fails for me on 10.6 ppc, which will be similar to 10.5 in this regard.

Reply:
Let me see if with these fixes it works. Will update in few min.

Reply:
Okay, thanks. Note that we can't use `vfork` on macos. So it must be a leftover from the old code when we did.